app/src/main/java/com/github/sdp/tarjetakuna/ui/profile/ProfileFragment.kt

```diff
@@ -1,50 +1,79 @@
 package com.github.sdp.tarjetakuna.ui.profile
 
+import android.content.Context
 import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.Button
 import android.widget.TextView
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.ViewModelProvider
 import com.github.sdp.tarjetakuna.MainActivity
+import com.github.sdp.tarjetakuna.R
 import com.github.sdp.tarjetakuna.databinding.FragmentProfileBinding
 
 class ProfileFragment : Fragment() {
 
-private var _binding: FragmentProfileBinding? = null
-  // This property is only valid between onCreateView and
-  // onDestroyView.
-  private val binding get() = _binding!!
-
-  override fun onCreateView(
-    inflater: LayoutInflater,
-    container: ViewGroup?,
-    savedInstanceState: Bundle?
-  ): View {
-    val profileViewModel =
+    private var _binding: FragmentProfileBinding? = null
+
+    // This property is only valid between onCreateView and
+    // onDestroyView.
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        val profileViewModel =
             ViewModelProvider(this).get(ProfileViewModel::class.java)
 
-    _binding = FragmentProfileBinding.inflate(inflater, container, false)
-    val root: View = binding.root
+        _binding = FragmentProfileBinding.inflate(inflater, container, false)
+        val root: View = binding.root
+        val sharedPref = requireActivity().getSharedPreferences(
+            "com.github.sdp.tarjetakuna",
+            Context.MODE_PRIVATE
+        )
 
-    val textView: TextView = binding.textProfile
-    profileViewModel.text.observe(viewLifecycleOwner) {
-      textView.text = it
-    }
+        fun sharedPrefEntryInitiate(textView: TextView, key: String, hint: String) {
+            // Not using a viewmodel for this as it is stored in sharedPref
+            textView.text = sharedPref.getString(key, hint)
+            textView.addTextChangedListener(object : TextWatcher {
+                override fun afterTextChanged(s: Editable?) {
+                    with(sharedPref.edit()) {
+                        putString(key, s.toString())
+                        apply()
+                    }
+                    val mainActivity = requireActivity() as MainActivity
+                    mainActivity.updateHeader()
```

Comment:
There exists also: onSharedPreferenceChangeListener which you can use directly in MainActivity so you don't have to create a MainActivity here.